lib/api/core/entrypoints/embedded/index.js

```diff
@@ -487,6 +487,9 @@ class EmbeddedEntryPoint extends EntryPoint {
    */
   newConnection (connection) {
     this.clients[connection.id] = connection;
+
+    this.kuzzle.emit('connection:new', JSON.parse(JSON.stringify(connection)));
```

Comment:
While I agree that we should make sure that the connection object cannot be altered, I feel that we should prevent unnecessary computation as much as possible.

I see two possibilities:
* make the ClientConnection constructor freezes itself (and its headers object). This needs a bit of investigation to make sure that this can be done without impacting the current code, but if possible, this would be the best option IMO
* emit only if there is [at least 1 listener)(https://nodejs.org/api/events.html#events_emitter_listenercount_eventname) to the connection:new event. This means extra-computation if there are at least one, unfortunately

Reply:
I choose to froze the `ClientConnection` instance.